react-hook-form-mui/src/Input.js

```diff
@@ -27,36 +27,34 @@ ErrorBoundary.propTypes = {
 	children: PropTypes.node,
 };
 
-const module = function({type, rhf}) {
-	switch (type) {
-		case 'array':
-			if (!rhf) throw new Error('`array` type is only supported via rhf. `rhf` prop must be set to true in order to use it.');
-			return require('./react-hook-form/InputArray').default;
-		case 'autocomplete':
-			return rhf ? require('./react-hook-form/Autocomplete').default : require('./forms/Autocomplete').default;
-		case 'select':
-			return rhf ? require('./react-hook-form/Select').default : require('./forms/Select').default;
-		case 'buttons':
-			return rhf ? require('./react-hook-form/ButtonGroup').default : require('./forms/ButtonGroup').default;
-		case 'multiLevelButtons':
-			return rhf ? require('./react-hook-form/MultiButtonGroup').default : require('./forms/ButtonGroup').default;
-		case 'inr':
-			return rhf ? require('./react-hook-form/CurrencyField').default : require('./forms/CurrencyField').default;
-		case 'checkbox':
-			return rhf ? require('./react-hook-form/CheckboxGroup').default : require('./forms/CheckboxGroup').default;
-		case 'selectchip':
-			return rhf ? require('./react-hook-form/SelectWithChip').default : require('./forms/SelectWithChip');
-		case 'slider':
-			return rhf ? require('./react-hook-form/Slider').default : require('./forms/Slider');
-	}
-	return rhf ?  require('./react-hook-form/TextField').default : require('./forms/TextField').default;
-};
-
-
 // eslint-disable-next-line react/no-multi-comp
 function Input({type, container, validate, label, rhf = true, components: {input, Field = input, Loader = LinearProgress, ...components} = {}, compact, ...rest}) {// eslint-disable-lineno-unused-varsno-unused-vars
 	const Container = container ? require('@mui/material/Grid').default : Fragment;
 	const containerProps = container ? {item: true, ...container} : {};
+	const module = ({type, rhf}) => {
```

Comment:
why is it inside input?

Reply:
Sir, It wasn't detecting Input component inside default export. It was returning "undefined".

Reply:
Sorry Sir, I tested it again and removed it from PR.